src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -47,19 +49,22 @@
 
     private ObservableList<TurboIssue> issuesToDisplay = null;
 
-    protected HBox panelMenuBar;
-    protected Text nameText;
-    protected HBox nameBox;
-    protected HBox menuBarNameArea;
-    protected HBox menuBarCloseButton;
-    protected Label renameButton;
-    protected Label closeButton;
-    protected PanelNameTextField renameTextField;
+    public HTPanelMenuBar panelMenuBar;
+    private Text nameText;
+    public HBox nameBox;
+    public HBox menuBarNameArea;
+    public HBox menuBarCloseButton;
+    public HBox menuBarConfirmButton;
+    public HBox menuBarUndoButton;
+    public Label renameButton;
+    public Label closeButton;
```

Comment:
I would say that the point of the menu bar class is exactly to abstract away details like CSS classes, what child objects are involved, etc. from FilterPanel.

FilterPanels are mainly concerned with filtering (dragging is related as its effects depend on the filter). The ability to be named is somewhat orthogonal, so we should try to keep it separate and abstract. It might even make sense in future to move it up to AbstractPanel, if we end up having panels with names that don't filter issues. Keeping things decoupled will help with that.

Anyway, if the menu bar component still turns out to be messy, it should be broken down further.